### PR TITLE
Clean up the canvas session DragState

### PR DESCRIPTION
**Pull request by balazsbajorics**
**Problem:**
We have a new implementation of the strategy proposal on the branch `refactor/interactions-proposal`. We soon want to merge that branch to master.

This PR has two goals:
1. remove the now obsolete dragState-based strategy implementation, to de-clutter master and make the way for the new strategies
2. As much as possible, make the new Strategy PR zero-based. Meaning I removed some files that the new PR will have to re-introduce, instead of slightly modifying (prototype quality) code that was currently on master.

**Commit Details:**
- Removing SelectModeCanvasSession
- Remove all changes made to dragState
- remove Commands
- Remove the example controls and strategies
- Remove the strategy indicator

### CODE REVIEW

**Review by gbalint — APPROVED** [review]

File: editor/src/components/canvas/controls/select-mode/select-mode-hooks.tsx [hunk 1]
```diff
@@ -312,9 +311,7 @@ function useStartCanvasSession(): (
       }
 
       dispatch([
-        CanvasActions.createDragState(
-          startNewSelectModeCanvasSession(start, { type: 'BOUNDING_AREA' }),
-        ),
+        // TODO actually dispatch a canvas interaction session
```
Comment (Rheeseyb) [inline editor/src/components/canvas/controls/select-mode/select-mode-hooks.tsx:314]: If we're making the strategies PR zero based then the whole of `useStartCanvasSession` should be deleted

Comment (balazsbajorics) [reply editor/src/components/canvas/controls/select-mode/select-mode-hooks.tsx:314]: fair point

File: editor/src/core/performance/performance-regression-tests.spec.tsx [hunk 2]
```diff
@@ -64,7 +64,7 @@ describe('React Render Count Tests -', () => {
 
     const renderCountAfter = renderResult.getNumberOfRenders()
     // if this breaks, GREAT NEWS but update the test please :)
-    expect(renderCountAfter - renderCountBefore).toMatchInlineSnapshot(`493`)
+    expect(renderCountAfter - renderCountBefore).toMatchInlineSnapshot(`483`)
```
Comment (Rheeseyb) [inline editor/src/core/performance/performance-regression-tests.spec.tsx:67]: Why did the render counts go down? The only new rendered elements were behind a flag that was disabled by default, so this is ... interesting

**Review by Rheeseyb — APPROVED** [review]

**Review by enidemi — APPROVED** [review]

File: editor/src/components/canvas/controls/select-mode/select-mode-hooks.tsx (hunk 1 above)
Comment (balazsbajorics) [reply editor/src/components/canvas/controls/select-mode/select-mode-hooks.tsx]: done! I actually removed the feature flag entirely

File: editor/src/core/performance/performance-regression-tests.spec.tsx (hunk 2 above)
Comment (balazsbajorics) [reply editor/src/core/performance/performance-regression-tests.spec.tsx:67]: the CanvasStrategyIndicator was not behind a flag – might be that, or maybe some other component that I removed now?

Comment (Rheeseyb) [reply editor/src/core/performance/performance-regression-tests.spec.tsx:67]: Ah it was probably that then

**Pull request merged**